core/src/main/scala/kafka/api/ApiVersion.scala

```diff
@@ -24,16 +24,25 @@ package kafka.api
  *
  * Note that the ID we initialize for each version is important.
  * We consider a version newer than another, if it has a higher ID (to avoid depending on lexicographic order)
+ * 
+ * If there is a draft protocol version between protocols of two official releases. The suffix "-IV#" will be added to
+ * help users who are running on trunk upgrade. For example:
+ * 1. Kafka 0.9.0 is released
+ * 2. After that some protocol change are made and will be released in 0.10.0. The version will be named 0.10.0-IV0
+ *    (IV stands for internal version)
+ * 3. When Kafka 0.10.0 is released. The official version 0.10.0 will be the same as the last internal version.
```

Comment:
Reworded this a bit. See if it's clearer.

Since the api protocol may change more than once within the same release, to facilitate people deploying code from trunk, we introduce internal versions since 0.10.0. For example, the first time that we introduce a version change in 0.10.0, we will add a config value "0.10.0-IV0" and a corresponding case object KAFKA_0_10_0-IV0. We will also add a config value "0.10.0" that will be mapped to the latest internal version object, which is KAFKA_0_10_0-IV0. When we change the protocol a second time while developing 0.10.0, we will add a new config value "0.10.0-IV1" and a corresponding case object KAFKA_0_10_0-IV1. We will change the config value "0.10.0" to map to the latest internal version object KAFKA_0_10_0-IV1. Config value of "0.10.0-IV0" is still mapped to KAFKA_0_10_0-IV0. This way, if people are deploying from trunk, they can use "0.10.0-IV0" and "0.10.0-IV1" to upgrade one internal version at a time. For most people who just want to use released version, they can use "0.10.0" when upgrading to 0.10.0 release.